Source/Client.swift

```diff
@@ -126,6 +128,9 @@ extension Client {
                 }
 
                 return response
+            } else if data.count == 0 {
+                //no data received, connection got broken
+                throw ClientError.brokenConnection
```

Comment:
I think we can find a better name for this.. something like `connectionResetByPeer` or brokenPipe`?

https://github.com/Zewo/POSIX/blob/master/Sources/POSIX.swift#L100

https://github.com/Zewo/POSIX/blob/master/Sources/POSIX.swift#L65

Also.. I think `receive` should be responsible to throw this error. is it not throwing? strange..